File: app/src/main/java/ru/telematica/casco2go/model/TripData.java
package ru.telematica.casco2go.model;

import java.util.Calendar;
import java.util.Locale;

/**
 * Created by m.sidorov on 29.04.2018.
 */

public class TripData {

    private int carPrice;
    private Calendar startTime;
    private Calendar finishTime;
    private boolean started = false;
    private boolean finished = false;

    public Calendar getStartTime(){
        return startTime;
    }

    public Calendar getFinishTime(){
        return finishTime;
    }

    public int getCarPrice(){
        return carPrice;
    }

    public boolean isStarted(){
        return started;
    }

    public boolean isFinished(){
        return finished;
    }

    // Время поездки в мс
    public long getTripTime(){
        if (!isStarted()) {
            return 0;
        } else
        if (!isFinished()){
            return System.currentTimeMillis() - startTime.getTimeInMillis();
        } else {
            return finishTime.getTimeInMillis() - startTime.getTimeInMillis();
        }
    }

    public int getGpsLevel(){
        return 98;
    }


    public TripData(int carPrice){
        this.carPrice = carPrice;
        this.started = false;
        this.finished = false;
    }

    public TripData start(){
        this.started = true;
        this.startTime = Calendar.getInstance();
        return this;
    }

    public TripData finish(){
        this.finished = true;
        this.finishTime = Calendar.getInstance();
        return this;
    }

}

File: app/src/main/java/ru/telematica/casco2go/model/TripData.kt
package ru.telematica.casco2go.model

import ru.telematica.casco2go.repository.ConfigRepository
import ru.telematica.casco2go.ui.fragments.FinishTripFragment
import ru.telematica.casco2go.utils.isNull
import java.util.*

/**
 * Created by m.sidorov on 29.04.2018.
 */
class TripData {

    var startTime: Long = 0
        private set

    var finishTime: Long = 0
        private set

    var started = false
        private set

    var finished = false
        private set

    var scoringData: ScoringData = ScoringData()
        private set

    val tripTime: Long
        get() {
            if (started){
                if (finished)
                    return finishTime - startTime
                else
                    return System.currentTimeMillis() - startTime
            } else
                return 0;
        }

    var gpsLevel: Int = 100

    @Synchronized
    fun start(): TripData {
        started = true
        finished = false;
        startTime = System.currentTimeMillis()
        finishTime = 0
        return this
    }

    @Synchronized
    fun finish(scoringData: ScoringData): TripData {
        finishTime = System.currentTimeMillis()
        finished = true
        this.scoringData = scoringData
        return this
    }

}

